### PR TITLE
Fix "module" external helpers output

### PR DESCRIPTION
**Pull request by loganfsmyth**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues             | Fixes #6367 
| Patch: Bug Fix?          | 
| Major: Breaking Change?  | 
| Minor: New Feature?      | 
| Tests Added/Pass?        | 
| Spec Compliancy?         | 
| License                  | MIT
| Doc PR                   | 
| Any Dependency Changes?  | 

This is an alternative to https://github.com/babel/babel/pull/6367 that I tries to keep things a little simpler.

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
I actually prefer this PR, since it removes the special-casing for keyword helpers.

File: packages/babel-core/src/tools/build-external-helpers.js [hunk 1]
```diff
@@ -56,63 +56,38 @@ function buildGlobal(namespace, builder) {
...
-      ]),
-      [],
-    );
-  });
+  const refs = buildHelpers(body, null, whitelist);
+
+  body.unshift(
+    t.exportNamedDeclaration(
+      null,
+      Object.keys(refs).map(name => {
```
Comment (Andarist) [inline packages/babel-core/src/tools/build-external-helpers.js]: havent thought about that, nice!

**Comment by Andarist** [comment]
Also I have added [this if statement](https://github.com/babel/babel/pull/6367/files#diff-3d02fe9d8c28472adc1e34fd7bcfd2c1R61) in order to handle exported identifiers (such as `temporalUndefined`) more gracefully [here](https://github.com/Andarist/babel/blob/aa1b8737fdbe7b439f06024a0f1e1d917df5d0cd/packages/babel-helpers/src/index.js#L175) - while it worked because `exportName` was undefined in that case and therefore different than expectec helper name (`id.name`), it could have been a little bit by luck and not sure if the original author meant it that way. Imho handling Identifier case there makes things more explicit and therefore better.

**Comment by loganfsmyth** [comment]
@Andarist Ahh, good point about `typeof`. I'll fix that.

I'm not sure I follow what you've saying about `temporalUndefined` though.

**Comment by Andarist** [comment]
This is perfect!

**Pull request merged**